File: app/src/main/java/com/inspiringteam/xchange/ui/quakes/QuakesAdapter.java
package com.inspiringteam.xchange.ui.quakes;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

import com.inspiringteam.xchange.R;

import java.util.List;

import static com.google.common.base.Preconditions.checkNotNull;

final class QuakesAdapter extends BaseAdapter{
    private List<QuakeItem> mQuakes;

    public QuakesAdapter(List<QuakeItem> quakes) {
        setList(quakes);
    }

    public void replaceData(List<QuakeItem> quakes) {
        setList(quakes);
        notifyDataSetChanged();
    }

    private void setList(List<QuakeItem> quakes) {
        mQuakes = checkNotNull(quakes);
    }

    @Override
    public int getCount() {
        return mQuakes.size();
    }

    @Override
    public QuakeItem getItem(int i) {
        return mQuakes.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        View rowView = view;
        QuakeItemViewHolder viewHolder;

        if (rowView == null) {
            LayoutInflater inflater = LayoutInflater.from(viewGroup.getContext());
            rowView = inflater.inflate(R.layout.quake_item, viewGroup, false);

            viewHolder = new QuakeItemViewHolder(rowView);
            rowView.setTag(viewHolder);
        } else {
            viewHolder = (QuakeItemViewHolder) rowView.getTag();
        }

        final QuakeItem quakeItem = getItem(i);
        viewHolder.bindItem(quakeItem);

        return rowView;
    }
}

File: app/src/main/java/com/inspiringteam/xchange/ui/quakes/QuakesAdapter.kt
package com.inspiringteam.xchange.ui.quakes

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import com.inspiringteam.xchange.R
import java.util.*

internal class QuakesAdapter(private var quakes: MutableList<QuakeItem>) : BaseAdapter() {

    fun replaceData(quakes: MutableList<QuakeItem>) {
        this.quakes = quakes
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return quakes.size
    }

    override fun getItem(i: Int): QuakeItem {
        return quakes[i]
    }

    override fun getItemId(i: Int): Long {
        return i.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var rowView = convertView
        val viewHolder: QuakeItemViewHolder
        if (rowView == null) {
            val inflater = LayoutInflater.from(parent?.context)
            rowView = inflater.inflate(R.layout.quake_item, parent, false)
            viewHolder = QuakeItemViewHolder(rowView)
            rowView.tag = viewHolder
        } else
            viewHolder = rowView.tag as QuakeItemViewHolder

        val quakeItem = getItem(position)
        viewHolder.bindItem(quakeItem)
        return rowView!!
    }

}